app/javascript/components/ui/Table.tsx

```diff
@@ -0,0 +1,168 @@
+
+const TableContext = React.createContext<{
+  headerLabels?: (string | null)[];
+  setHeaderLabels: (headerLabels: (string | null)[]) => void;
+}>({ setHeaderLabels: () => {} });
+const useTable = () => assertDefined(React.useContext(TableContext), "useTable must be used within a Table");
+
+export const Table = React.forwardRef<HTMLTableElement, React.HTMLAttributes<HTMLTableElement>>(
+  ({ className, children, ...props }, ref) => {
+    const [headerLabels, setHeaderLabels] = React.useState<(string | null)[]>([]);
+    const contextValue = React.useMemo(() => ({ headerLabels, setHeaderLabels }), [headerLabels, setHeaderLabels]);
+    return (
+      <TableContext.Provider value={contextValue}>
+        <table
+          ref={ref}
+          className={classNames(
+            "grid w-full border-spacing-0 gap-4 lg:table lg:rounded-sm lg:border lg:border-border",
+            className,
+          )}
+          {...props}
+        >
+          {children}
+        </table>
+      </TableContext.Provider>
+    );
+  },
+);
+Table.displayName = "Table";
+
+export const TableCaption = ({ className, children, ...props }: React.HTMLAttributes<HTMLTableCaptionElement>) => (
+  <caption className={classNames("block text-left text-base text-xl lg:mb-4 lg:table-caption", className)} {...props}>
+    {children}
+  </caption>
+);
+
+export const TableHeader = ({ className, children, ...props }: React.HTMLAttributes<HTMLTableSectionElement>) => (
+  <thead className={classNames("hidden lg:table-header-group", className)} {...props}>
+    {children}
+  </thead>
+);
+
+export const TableBody = ({ className, children, ...props }: React.HTMLAttributes<HTMLTableSectionElement>) => (
+  <tbody className={classNames("contents lg:table-row-group lg:rounded-sm", className)} {...props}>
+    {children}
+  </tbody>
+);
+
+export const TableFooter = ({ className, children, ...props }: React.HTMLAttributes<HTMLTableSectionElement>) => (
+  <tfoot className={classNames("contents font-bold lg:table-footer-group", className)} {...props}>
+    {children}
+  </tfoot>
+);
+
+export const TableRow = ({
+  className,
+  selected,
+  children,
+  ...props
+}: Omit<React.HTMLAttributes<HTMLTableRowElement>, "aria-selected"> & { selected?: boolean; footer?: boolean }) => {
+  const { headerLabels: headers, setHeaderLabels: setHeaders } = useTable();
+
+  React.useEffect(() => {
+    const isHeaderRow = React.Children.toArray(children).every(
+      (child) => React.isValidElement(child) && child.type === TableHead,
+    );
+    if (!isHeaderRow) return;
+
+    setHeaders(
+      React.Children.toArray(children).map<string | null>((child) => {
+        if (React.isValidElement(child) && child.type === TableHead) {
+          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+          const props = child.props as React.ComponentProps<typeof TableHead>;
+          return typeof props.children === "string" ? props.children : null;
```

Comment:
Do we need this check here? Ideally if we have some simple HTML that would get rendered in the mobile view as well

Reply:
True, I still had the old `aria-label` attribute in my mind for some reason 🤦